compiler/rustc_mir_build/src/thir/pattern/check_match.rs

```diff
@@ -479,12 +479,30 @@ impl<'p, 'tcx> MatchVisitor<'_, 'p, 'tcx> {
             AdtDefinedHere { adt_def_span, ty, variants }
         };
 
+        // Emit an extra note if the first uncovered witness is
+        // visibly uninhabited anywhere in the current crate.
+        let witness_1_is_privately_uninhabited =
+            if cx.tcx.features().exhaustive_patterns
+                && let Some(witness_1) = witnesses.get(0)
+                && let ty::Adt(adt, substs) = witness_1.ty().kind()
+                && adt.is_enum()
+                && let Constructor::Variant(variant_index) = witness_1.ctor()
+            {
+                let variant = adt.variant(*variant_index);
+                let inhabited = variant.inhabited_predicate(cx.tcx, *adt).subst(cx.tcx, substs);
+                assert!(inhabited.apply(cx.tcx, cx.param_env, cx.module));
+                !inhabited.apply_ignore_module_in_crate(cx.tcx, cx.param_env)
```

Comment:
I'm slightly confused about the logic here: from reading the doc-comments of these two functions, don't they return the same thing for any module that is not in the current crate?

Reply:
My understanding of what we want the logic to do is: if the first witness is a variant that is uninhabited *in its native crate/module* (or in fact anywhere at all is enough) then we emit a note. It shouldn't be about the current crate